core/src/org/javarosa/xpath/expr/XPathFuncExpr.java

```diff
@@ -145,6 +145,13 @@ public Object eval(DataInstance model, EvaluationContext evalContext) {
 
         Hashtable funcHandlers = evalContext.getFunctionHandlers();
 
+
+        //check for custom handler, use this if it exists.
+        IFunctionHandler handler = (IFunctionHandler)funcHandlers.get(name);
+        if (handler != null) {
+            return evalCustomFunction(handler, argVals, evalContext);

```

Comment:
Question: This goes by name, what is the custom function handler is overriding a multiple argument function. 

IE: 

custom function = if(a, b, c, d) (so a ridiculous 4-if argument)

We should probably have another return path from evalCustomFunction() which says "Actually, I have a named function for this, but no matching argument # signatures" that still lets it fall back down to the default functions.